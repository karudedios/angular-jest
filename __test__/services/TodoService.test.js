import * as R from 'ramda';

describe("TodoService", function() {
  let service;

  beforeEach(() => {

    angular.mock.module('jest-webpack-app.services');

  });

  beforeEach(inject(function(TodoService) {

    service = TodoService;

  }));

  afterEach(function() {

    service.__teardown();

  });

  it('should return a new instance of a Todo object on `new` call', async function() {

    const data = { title: 'Hello' };

    const todo = await service.new(data);

    expect(todo.id).toBeDefined();
    expect(todo.completed).toEqual(false);
    expect(todo.title).toEqual(data.title);

  });

  it('should store a new todo on the "datasource" collection', async function() {

    const data = { title: 'Hello' };

    const todo = await service.new(data);
    const todos = await service.all();

    expect(todos.length).toEqual(1);
    expect(R.last(todos)).toEqual(todo);

  });

  it('should allow you to update an existing todo', async function() {

    const data = { title: 'Hello' };
    const titleLens = R.lensProp('title');

    const todo = await service.new(data);
    const updatedTodo = await service.update(todo.id, R.set(titleLens, "Hello 2", todo));

    expect(updatedTodo.id).toEqual(todo.id);
    expect(updatedTodo.title).toEqual('Hello 2');
    expect(updatedTodo.completed).toEqual(todo.completed);

  });

  it('should allow you to delete existing todos', async function() {

    const data = { title: 'Hello '};

    const todo = await service.new(data);

    const deleted = await service.delete(todo.id);

    const allTodos = await service.all();

    expect(deleted).toEqual(true);
    expect(allTodos.length).toEqual(0);

  });

  it('should throw an error if todo id does not exist when trying to update', async function() {

    try {

      await service.update(1, {});

    } catch(e) {

      expect(e.message).toEqual('Todo not found');

    }

  });

  it('should throw an error if todo id does not exist when trying to delete', async function() {

    try {

      await service.delete(1);

    } catch(e) {

      expect(e.message).toEqual('Todo not found');

    }

  });

});
