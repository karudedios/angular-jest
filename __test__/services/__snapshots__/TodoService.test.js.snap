// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoService should allow you to delete existing todos 1`] = `Array []`;

exports[`TodoService should allow you to update an existing todo 1`] = `
Object {
  "completed": false,
  "id": 8,
  "title": "Hello 2",
}
`;

exports[`TodoService should return a new instance of a Todo object on \`new\` call 1`] = `
Todo {
  "completed": false,
  "id": 6,
  "title": "Hello",
}
`;

exports[`TodoService should store a new todo on the "datasource" collection 1`] = `
Todo {
  "completed": false,
  "id": 7,
  "title": "Hello",
}
`;

exports[`TodoService should throw an error if todo id does not exist when trying to delete 1`] = `[Error: Todo not found]`;

exports[`TodoService should throw an error if todo id does not exist when trying to update 1`] = `[Error: Todo not found]`;
